persistence/OrderPersistence.js

```diff
@@ -28,6 +28,12 @@ class OrderPersistence extends Persistence {
         this.assertExists(order, orderId, Order.entityName);
         await order.destroy();
     }
+    
+    async getByProductId(productId) {
+        const orders = await Order.findAll({ where: { productId } });
+        this.assertExists(orders, productId, Order.entityName);
```

Comment:
You are passing an order and a productId to the assertExists?

This will fail if the order doesn't exist but the product exists and the error message will be:

"Order with ID <the id of the product?> not found